--- a/src/collection/sqlcollection/CMakeLists.txt
+++ b/src/collection/sqlcollection/CMakeLists.txt
@@ -74,9 +74,8 @@
 )
 
 if(NOT WIN32)
-    target_link_libraries(amarok_collection-sqlcollection crypto ssl)
     if(NOT APPLE)
-        target_link_libraries(amarok_collection-sqlcollection crypt pthread)
+        target_link_libraries(amarok_collection-sqlcollection crypt pthread dl)
     endif(NOT APPLE)
 endif(NOT WIN32)
 
