From: Modestas Vainius <modax@debian.org>
Subject: Fix building on armel
Origin: backport, commit:8e5fe76cfefa5a17e969953f4f8249617c77a561
Last-Update: 2011-07-24
Applied-Upstream: 2.4.3

--- a/src/dynamic/biases/PartBias.cpp
+++ b/src/dynamic/biases/PartBias.cpp
@@ -568,7 +568,7 @@ Dynamic::PartBias::changeBiasWeight( int
     if( biasNum != 0 )
     {
         qreal oldV = m_weights.at(0);
-        qreal newV = qBound( qreal( 0.0 ), 1.0 - (sum - oldV), qreal( 1.0 ) );
+        qreal newV = qBound<qreal>( qreal( 0.0 ), 1.0 - (sum - oldV), qreal( 1.0 ) );
         m_weights[0] = newV;
 
         sum = sum - oldV + newV;
